app/components/todos-list.js

```diff
@@ -2,25 +2,18 @@ import React, {Component} from 'react';
 import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 import {delTodo, checkTodo} from '../actions/index';
+import TodoItem from './todo-item';
 
 class TodoList extends Component{
 
-  createListItems(){
-    return this.props.todos.map((todo) =>{
-    return (
-      <li key={todo.id}>
-        <input type="checkbox" checked={todo.checked} onChange={() => this.props.checkTodo(todo)}/>
-        <span>{todo.title} - </span>
-        <span onClick={() => this.props.delTodo(todo)} style={{color: 'red'}}>X</span>
-      </li>
-    );
-  });
-  }
-
   render(){
     return (
       <ul>
-        {this.createListItems()}
+        {this.props.todos.map((todo) =>{
+        return (
+          <TodoItem key={todo.id} todo={todo} del={this.props.delTodo} check={this.props.checkTodo}/>
```

Comment:
These actions come from the actions file. There is no need to inject them from the parent. Move them to the TodoItem and use connect / bindActions there. It will play much better with Redux and will be clearer to read (ex: whats del vs. delTodo? Is it the action or a method that calls dispatch? I know it will call dispatch with that method because of the bindActions thing, but it took me a while to figure it out).